### PR TITLE
Functional tests - Add new test to check 'BO > Product V2 > Combination tab'

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop 
| Description?      | Add new test to check 'BO > Product V2 > Combination tab'
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | no need
| UI Tests          | Nightly => https://github.com/nesrineabdmouleh/testing_pr/actions/runs/6262240998
| Fixed issue or discussion?     | no
| Related PRs       | no
| Sponsor company   |

### CODE REVIEW

**Review by boubkerbribri — DISMISSED** [review]

File: tests/UI/pages/BO/catalog/productsV2/add/combinationsTab.ts [hunk 1]
```diff
@@ -412,6 +511,36 @@ class CombinationsTab extends BOBasePage {
...
+  }
+
+  /**
+   * Get combination name from modal
+   * @param page {Page} Browser tab
+   * @returns {Promise<string>}
+   */
+  async getCombinationNameFromModal(page: Page): Promise<string> {
+    const combinationFrame: Frame | null = await page.frame({url: /sell\/catalog\/products-v2\/combinations/gmi});
+    await expect(combinationFrame).to.be.not.null;
```
Comment (Progi1984) [inline tests/UI/pages/BO/catalog/productsV2/add/combinationsTab.ts]:
```suggestion
    const combinationFrame: Frame | null = page.frame({url: /sell\/catalog\/products-v2\/combinations/gmi});
    expect(combinationFrame).to.be.not.eq(null);
```

File: tests/UI/campaigns/productV2/functional/11_combinationTab.ts [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+      expect(pageTitle).to.contains(createProductsPage.pageTitle);
+    });
+  });
+
+  describe('Delete product', async () => {
+    it('should delete product', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'deleteProduct', baseContext);
+
+      const createProductMessage = await createProductsPage.deleteProduct(page);
+      await expect(createProductMessage).to.equal(productsPage.successfulDeleteMessage);
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/11_combinationTab.ts]:
```suggestion
      expect(createProductMessage).to.equal(productsPage.successfulDeleteMessage);
```

File: tests/UI/campaigns/productV2/functional/11_combinationTab.ts [hunk 3]
```diff
@@ -0,0 +1,426 @@
...
+    });
+
+    it('should check the label of out of stock', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'selectCombination6', baseContext);
+
+      const availabilityLabel = await foProductPage.getProductAvailabilityLabel(page);
+      await expect(availabilityLabel).to.contains('Out of stock');
+
+      const isVisible = await foProductPage.isAddToCartButtonEnabled(page);
+      await expect(isVisible).to.be.true;
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/11_combinationTab.ts]:
```suggestion
      expect(isVisible).to.be.equal(true);
```

File: tests/UI/campaigns/productV2/functional/11_combinationTab.ts [hunk 4]
```diff
@@ -0,0 +1,426 @@
...
+        },
+        {
+          name: 'color',
+          value: 'Taupe',
+        },
+      ];
+      await foProductPage.selectAttributes(page, 1, firstCombination);
+
+      const isVisible = await foProductPage.isAddToCartButtonEnabled(page);
+      await expect(isVisible).to.be.true;
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/11_combinationTab.ts]:
```suggestion
      expect(isVisible).to.be.equal(true);
```

File: tests/UI/campaigns/productV2/functional/11_combinationTab.ts [hunk 5]
```diff
@@ -0,0 +1,426 @@
...
+    it('should preview product', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'previewProduct4', baseContext);
+
+      // Click on preview button
+      page = await createProductsPage.previewProduct(page);
+
+      await foProductPage.changeLanguage(page, 'en');
+
+      const pageTitle = await foProductPage.getPageTitle(page);
+      await expect(pageTitle).to.contains(newProductData.name);
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/11_combinationTab.ts]:
```suggestion
      expect(pageTitle).to.contains(newProductData.name);
```

File: tests/UI/campaigns/productV2/functional/11_combinationTab.ts [hunk 6]
```diff
@@ -0,0 +1,426 @@
...
+
+      const isVisible = await foProductPage.isAddToCartButtonEnabled(page);
+      await expect(isVisible).to.be.false;
+    });
+
+    it('should check the product availability label', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'selectCombination4', baseContext);
+
+      const availabilityLabel = await foProductPage.getProductAvailabilityLabel(page);
+      await expect(availabilityLabel).to.contains('Product available');
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/11_combinationTab.ts]:
```suggestion
      expect(availabilityLabel).to.contains('Product available');
```

File: tests/UI/campaigns/productV2/functional/11_combinationTab.ts [hunk 7]
```diff
@@ -0,0 +1,426 @@
...
+        },
+        {
+          name: 'color',
+          value: 'Taupe',
+        },
+      ];
+      await foProductPage.selectAttributes(page, 1, firstCombination);
+
+      const isVisible = await foProductPage.isAddToCartButtonEnabled(page);
+      await expect(isVisible).to.be.false;
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/11_combinationTab.ts]:
```suggestion
      expect(isVisible).to.be.equal(false);
```

File: tests/UI/campaigns/productV2/functional/11_combinationTab.ts [hunk 8]
```diff
@@ -0,0 +1,426 @@
...
+    it('should preview product', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'previewProduct3', baseContext);
+
+      // Click on preview button
+      page = await createProductsPage.previewProduct(page);
+
+      await foProductPage.changeLanguage(page, 'en');
+
+      const pageTitle = await foProductPage.getPageTitle(page);
+      await expect(pageTitle).to.contains(newProductData.name);
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/11_combinationTab.ts]:
```suggestion
      expect(pageTitle).to.contains(newProductData.name);
```

File: tests/UI/campaigns/productV2/functional/11_combinationTab.ts [hunk 9]
```diff
@@ -0,0 +1,426 @@
...
+      });
+    });
+
+    it('should click on edit default behaviour link', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'editBehaviour', baseContext);
+
+      page = await combinationsTab.clickOnEditDefaultBehaviourLink(page);
+
+      const pageTitle = await productSettings.getPageTitle(page);
+      await expect(pageTitle).to.contains(productSettings.pageTitle);
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/11_combinationTab.ts]:
```suggestion
      expect(pageTitle).to.contains(productSettings.pageTitle);
```

File: tests/UI/campaigns/productV2/functional/11_combinationTab.ts [hunk 10]
```diff
@@ -0,0 +1,426 @@
...
+      });
+
+      it('should go back to BO', async function () {
+        await testContext.addContextItem(this, 'testIdentifier', `goBackToBO${index}`, baseContext);
+
+        // Go back to BO
+        page = await foProductPage.closePage(browserContext, page, 0);
+
+        const pageTitle = await createProductsPage.getPageTitle(page);
+        await expect(pageTitle).to.contains(createProductsPage.pageTitle);
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/11_combinationTab.ts]:
```suggestion
        expect(pageTitle).to.contains(createProductsPage.pageTitle);
```

File: tests/UI/campaigns/productV2/functional/11_combinationTab.ts [hunk 11]
```diff
@@ -0,0 +1,426 @@
...
+          },
+          {
+            name: 'color',
+            value: 'White',
+          },
+        ];
+        await foProductPage.selectAttributes(page, 1, firstCombination);
+
+        const isVisible = await foProductPage.isAddToCartButtonEnabled(page);
+        await expect(isVisible).to.eq(test.args.isAddToCartButtonVisible);
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/11_combinationTab.ts]:
```suggestion
        expect(isVisible).to.be.equal(test.args.isAddToCartButtonVisible);
```

File: tests/UI/campaigns/productV2/functional/11_combinationTab.ts [hunk 12]
```diff
@@ -0,0 +1,426 @@
...
+      it('should preview product', async function () {
+        await testContext.addContextItem(this, 'testIdentifier', `previewProduct${index}`, baseContext);
+
+        // Click on preview button
+        page = await createProductsPage.previewProduct(page);
+
+        await foProductPage.changeLanguage(page, 'en');
+
+        const pageTitle = await foProductPage.getPageTitle(page);
+        await expect(pageTitle).to.contains(newProductData.name);
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/11_combinationTab.ts]:
```suggestion
        expect(pageTitle).to.contains(newProductData.name);
```

File: tests/UI/campaigns/productV2/functional/11_combinationTab.ts [hunk 13]
```diff
@@ -0,0 +1,426 @@
...
+
+      const combinationName = await combinationsTab.getCombinationNameFromModal(page);
+      await expect(combinationName).to.equal('Size - S, Color - Gray');
+    });
+
+    it('should close the modal', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'closeModal', baseContext);
+
+      const isModalVisible = await combinationsTab.closeEditCombinationModal(page);
+      await expect(isModalVisible).to.be.false;
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/11_combinationTab.ts]:
```suggestion
      expect(isModalVisible).toto.be.equal(false);
```

File: tests/UI/campaigns/productV2/functional/11_combinationTab.ts [hunk 14]
```diff
@@ -0,0 +1,426 @@
...
+      await expect(combinationName).to.equal('Size - S, Color - Taupe');
+    });
+
+    it.skip('should click on previous combination button', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'clickOnPreviousButton', baseContext);
+
+      await combinationsTab.clickOnPreviousCombinationButton(page);
+
+      const combinationName = await combinationsTab.getCombinationNameFromModal(page);
+      await expect(combinationName).to.equal('Size - S, Color - Gray');
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/11_combinationTab.ts:217]:
```suggestion
      expect(combinationName).to.equal('Size - S, Color - Gray');
```

File: tests/UI/campaigns/productV2/functional/11_combinationTab.ts [hunk 15]
```diff
@@ -0,0 +1,426 @@
...
+    });
+
+    // @todo : https://github.com/PrestaShop/PrestaShop/issues/33965
+    it.skip('should click on next combination button', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'clickOnNextButton', baseContext);
+
+      await combinationsTab.clickOnNextCombinationButton(page);
+
+      const combinationName = await combinationsTab.getCombinationNameFromModal(page);
+      await expect(combinationName).to.equal('Size - S, Color - Taupe');
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/11_combinationTab.ts]:
```suggestion
      expect(combinationName).to.equal('Size - S, Color - Taupe');
```

**Review by Progi1984 — COMMENTED** [review]
Some comments


Please remove `await` before `expect` use.

**Review by boubkerbribri — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Comment by boubkerbribri** [comment]
THanks @nesrineabdmouleh

**Pull request merged**